state/controller_test.go

```diff
@@ -186,6 +186,29 @@ func (s *ControllerSuite) TestRemovingUnknownName(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, `unknown controller config setting "dr-worm"`)
 }
 
+func (s *ControllerSuite) TestUpdateControllerConfigAcceptEmptyStringSpace(c *gc.C) {
+	sp, err := s.State.AddSpace("ha-space", "", nil, false)
+	c.Assert(err, jc.ErrorIsNil)
+
+	m, err := s.State.AddMachine(state.UbuntuBase("12.10"), state.JobManageModel, state.JobHostUnits)
+	c.Assert(err, jc.ErrorIsNil)
+
+	addr := network.NewSpaceAddress("192.168.9.9")
+	addr.SpaceID = sp.Id()
+
+	c.Assert(m.SetProviderAddresses(addr), jc.ErrorIsNil)
+
+	err = s.State.UpdateControllerConfig(map[string]interface{}{
+		controller.JujuManagementSpace: "ha-space",
```

Comment:
I'd probably use the other space config, `JujuHASpace`, here just because it's the one we care about for the change, but it's not a big deal.

Reply:
Good catch, it was a mistake. Fixed it.